src/main/java/seedu/duke/Main.java

```diff
@@ -24,37 +103,67 @@ public static void main(String[] args) {
         WeekView weekView = new WeekView(startOfWeek, dateFormatter);
         TaskManager taskManager = new TaskManager();
 
+        boolean printWeek = true; // Flag to control printing of the week view
+        boolean inMonthView = false; // Flag to indicate if we are in month view mode
+
         while (true) {
-            weekView.printWeekView(taskManager);
+            if (printWeek) {
+                if (!inMonthView) {
+                    weekView.printWeekView(taskManager);
+                } else {
+                    weekView.printMonthView(taskManager);
+                }
+            }
+            printWeek = true; // Reset flag for the next iteration
             System.out.println("Enter 'next' for next week, 'prev' for previous week, 'add' to add a task, " +
-                    "'delete' to delete a task, or 'quit' to quit:");
+                    "'delete' to delete a task, 'month' to display the month view, " +
+                    "or 'quit' to quit:");
             String input = scanner.nextLine().trim().toLowerCase();
 
-            try {
-                switch (input) {
-                case "next":
+            switch (input) {
+            case "next":
+                if (inMonthView) {
+                    weekView.nextMonth();
+                } else {
                     weekView.nextWeek();
-                    break;
-                case "prev":
+                }
+                break;
+            case "prev":
+                if (inMonthView) {
+                    weekView.previousMonth();
+                } else {
                     weekView.previousWeek();
-                    break;
-                case "add":
-                    addManager(scanner, weekView, taskManager);
-                    break;
-                case "delete":
+                }
+                break;
+            case "add":
+                try {
+                    addManager(scanner, weekView, taskManager, inMonthView);
+                } catch (TaskManagerException e) {
+                    System.out.println(e.getMessage());
+                }
+                break;
+            case "delete":
+                try {
```

Comment:
We could perhaps rename the weekView instance of the WeekView class as it now more generally represents the current date view. This would help the code be more readable. Alternatively, we could also look into refactoring out the month related methods into its own class if we wish to maintain this naming